File: tests/__tests__/functions/return_array/return_array.js

import { expect, test } from '@jest/globals';
import { runFile } from '../../../helpers/testsHelper';

let pjs = runFile(import.meta.url, 'return_array.pas');
// insp(pjs.engine.tree);

test("write and writeln don't fail", () => {
  expect(pjs.getVarValue('e')).toBe(152)
});

File: tests/__tests__/functions/return_array/return_array.pas
var a: array [1 .. 5] of integer;
e: integer;

function setItem(): array [1 .. 5] of integer;
var  b: array [1 .. 5] of integer;
begin
    b[1] := 152;
    setItem := b;
end;

begin
   e := setItem()[1];
   writeln(e);
end.